activerecord/lib/active_record/connection_adapters/postgresql/database_statements.rb

```diff
@@ -163,7 +163,6 @@ def perform_query(raw_connection, sql, binds, type_casted_binds, prepare:, notif
             end
 
             verified!
-            handle_warnings(result)
```

Comment:
Was this just broken before? 😅 

Now that I'm looking at it, and at risk of furthering the side-track... is `execute` the right place to be clearing out the warning list? My first thought was maybe we should do it closer to the new `handle_warnings` callsite, but on second thought, maybe `handle_warnings` itself should just shift them out while it consumes them.

Reply:
> Was this just broken before?

No why? It's just that before it's the concrete adapters that would call it, and now it's done by the abstract adapter.

Reply:
It's passing `result` and the corresponding method seems to think it's getting `sql`

Reply:
Some implementation would take the `result`, other the `sql`.

Reply:
Ah, you are right. Yeah, the PG one was broken it seems.